titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeModule.java

```diff
@@ -77,6 +87,24 @@ public FixedIntervalTokenBucketConfiguration getGcUnknownPodsTokenBucketConfigur
         return factory.newProxy(FixedIntervalTokenBucketConfiguration.class, "titusMaster.kube.gcUnknownPodsTokenBucket");
     }
 
+    @Provides
+    @Singleton
+    public PodResourcePoolResolver getPodResourcePoolResolver(DirectKubeConfiguration configuration,
+                                                              Config config,
+                                                              ApplicationSlaManagementService capacityGroupService,
+                                                              TitusRuntime titusRuntime) {
+        return new PodResourcePoolResolverChain(Arrays.asList(
```

Comment:
minor: consider adding a `GpuResourcePoolResolver` into the chain to automatically assign jobs requesting GPUs to `RESOURCE_POOL_GPU_PREFIX` pools

Reply:
I have intentionally left GPU out of the picture for now. The way we handle them is complicated, and I want to address it in a separate PR.